src/components/base/Client.ts

```diff
@@ -0,0 +1,91 @@
+import * as googleProtobufStructPb from 'google-protobuf/google/protobuf/struct_pb'
+import { Client, baseApi, commonApi } from '../../main'
+import type { Base } from './Base'
+import { BaseServiceClient } from '../../gen/component/base/v1/base_pb_service.esm'
+import { promisify } from '../../utils'
+
+
+export class BaseClient implements Base {
+  private client: BaseServiceClient
+  private name: string
+
+  constructor (client: Client, name:string) {
```

Comment:
note to self / nit: our lint step should be catching these sort of minor issues (i.e. adding a space after `name:` in this case)

Reply:
hrm, I'm actually get a bunch of linting errors - curious why they aren't showing up in CI

Reply:
don't worry about this, [I'm making a separate branch to tighten up some of our linting rules](https://github.com/viamrobotics/viam-typescript-sdk/tree/update-lint-build). also, my errors were related to generating protobufs with another tool - I'll address that situation in the aforementioned branch as well.